dandi/validate_types.py

```diff
@@ -25,11 +105,20 @@ class Scope(Enum):
     DATASET = "dataset"
 
 
-@dataclass
-class ValidationResult:
+class ValidationResult(BaseModel):
     id: str
-    origin: ValidationOrigin
+
+    origin: Origin
+    """Origin of the validation result as validator and standard used in producing it"""
```

Comment:
```suggestion
    #: Origin of the validation result as validator and standard used in producing it
    origin: Origin
```

And likewise for other attributes.

Reply:
Using a triple-quoted string in this context will gain IDE support and other documentation tool support.